src/Castle.Core.Tests/DynamicProxy.Tests/ComInterfacesTests.cs

```diff
@@ -29,30 +29,32 @@ public class ComInterfacesTests:BasePEVerifyTestCase
 		[Test]
 		public void Marshal_Release_throws_when_called_with_IntPtr_Zero()
 		{
-			Assert.Throws<ArgumentNullException>(() => Marshal.Release(IntPtr.Zero));
+			Assert.Catch<ArgumentException>(() => Marshal.Release(IntPtr.Zero));
+			//           ^^^^^^^^^^^^^^^^^
+			// We're not going to be more precise because Mono and the CLR don't throw the same exception type.
+			// See https://github.com/mono/mono/issues/15853.
 		}
 
+		// NOTE: You may wonder why the following tests delegate their implementation to static methods
+		// in a private static class. This is because the method bodies contain references to ADODB.dll,
+		// which fails to load under Mono/Linux, causing test discovery to abort early. By moving these
+		// method bodies somewhere else, test discovery will succeed and we get a chance at specifying
+		// how the tests should run, e.g. using the `[Platform]` attributes.
```

Comment:
ADODB.dll is now gone, is this still accurate?

Reply:
Oh, you're absolutely right. I'll remove this before merging.